// @mantine/spotlight/cjs/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_spotlight_store = require("./spotlight.store.cjs");
const require_is_actions_group = require("./is-actions-group.cjs");
const require_SpotlightAction = require("./SpotlightAction.cjs");
const require_SpotlightActionsGroup = require("./SpotlightActionsGroup.cjs");
const require_SpotlightActionsList = require("./SpotlightActionsList.cjs");
const require_SpotlightEmpty = require("./SpotlightEmpty.cjs");
const require_SpotlightFooter = require("./SpotlightFooter.cjs");
const require_SpotlightRoot = require("./SpotlightRoot.cjs");
const require_SpotlightSearch = require("./SpotlightSearch.cjs");
const require_Spotlight = require("./Spotlight.cjs");
exports.Spotlight = require_Spotlight.Spotlight;
exports.SpotlightAction = require_SpotlightAction.SpotlightAction;
exports.SpotlightActionsGroup = require_SpotlightActionsGroup.SpotlightActionsGroup;
exports.SpotlightActionsList = require_SpotlightActionsList.SpotlightActionsList;
exports.SpotlightEmpty = require_SpotlightEmpty.SpotlightEmpty;
exports.SpotlightFooter = require_SpotlightFooter.SpotlightFooter;
exports.SpotlightRoot = require_SpotlightRoot.SpotlightRoot;
exports.SpotlightSearch = require_SpotlightSearch.SpotlightSearch;
exports.closeSpotlight = require_spotlight_store.closeSpotlight;
exports.createSpotlight = require_spotlight_store.createSpotlight;
exports.createSpotlightStore = require_spotlight_store.createSpotlightStore;
exports.isActionsGroup = require_is_actions_group.isActionsGroup;
exports.openSpotlight = require_spotlight_store.openSpotlight;
exports.spotlight = require_spotlight_store.spotlight;
exports.toggleSpotlight = require_spotlight_store.toggleSpotlight;
exports.useSpotlight = require_spotlight_store.useSpotlight;
