// @mantine/modals/cjs/events.cjs
"use client";
let _mantine_core = require("@mantine/core");
let _mantine_hooks = require("@mantine/hooks");
//#region packages/@mantine/modals/src/events.ts
const [useModalsEvents, createEvent] = (0, _mantine_core.createUseExternalEvents)("mantine-modals");
const openModal = (payload) => {
	const id = payload.modalId || (0, _mantine_hooks.randomId)();
	createEvent("openModal")({
		...payload,
		modalId: id
	});
	return id;
};
const openConfirmModal = (payload) => {
	const id = payload.modalId || (0, _mantine_hooks.randomId)();
	createEvent("openConfirmModal")({
		...payload,
		modalId: id
	});
	return id;
};
const openContextModal = (payload) => {
	const id = payload.modalId || (0, _mantine_hooks.randomId)();
	createEvent("openContextModal")({
		...payload,
		modalId: id
	});
	return id;
};
const closeModal = createEvent("closeModal");
const closeAllModals = createEvent("closeAllModals");
const updateModal = (payload) => createEvent("updateModal")(payload);
const updateContextModal = (payload) => createEvent("updateContextModal")(payload);
const modals = {
	open: openModal,
	close: closeModal,
	closeAll: closeAllModals,
	openConfirmModal,
	openContextModal,
	updateModal,
	updateContextModal
};
//#endregion
exports.closeAllModals = closeAllModals;
exports.closeModal = closeModal;
exports.modals = modals;
exports.openConfirmModal = openConfirmModal;
exports.openContextModal = openContextModal;
exports.openModal = openModal;
exports.updateContextModal = updateContextModal;
exports.updateModal = updateModal;
exports.useModalsEvents = useModalsEvents;
